pkg/adapterManager/manager2.go

```diff
@@ -0,0 +1,438 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapterManager
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	rpc "github.com/googleapis/googleapis/google/rpc"
+	"github.com/hashicorp/go-multierror"
+	"github.com/opentracing/opentracing-go"
+	tracelog "github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"istio.io/mixer/pkg/adapter"
+	adptTmpl "istio.io/mixer/pkg/adapter/template"
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/attribute"
+	cpb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/pool"
+	"istio.io/mixer/pkg/status"
+	"istio.io/mixer/pkg/template"
+)
+
+// RuntimeAction is the runtime representation of configured action - cpb.Action.
+// Configuration is processed to hydrate instance names to Instances.
+type RuntimeAction struct {
+	// HandlerName to dispatch instances to.
+	HandlerName string
+	// Instances to dispatch to the handler.
+	Instances []*cpb.Instance
+}
+
+// HandlerState represents a ready to use handler.
+type HandlerState struct {
+	// ready to use handler.
+	Handler adapter.Handler
+	// builder that produced the above handler.
+	Builder *adapter.BuilderInfo
+}
+
+// RuntimeState represents the current snapshot of the configuration database
+// and associated, initialized handlers.
+type RuntimeState interface {
+	// Resolve resolves configuration to a list of actions.
+	ResolveConfig(bag attribute.Bag, variety adptTmpl.TemplateVariety) ([]*RuntimeAction, error)
+
+	// Handler returns a ready to use handler
+	Handler(name string) (*HandlerState, bool)
+}
+
+// NewManager2 creates a new Manager2 dispatcher.
+func NewManager2(mapper expr.Evaluator, templateRepo template.Repository,
+	rt RuntimeState, gp *pool.GoroutinePool) Dispatcher {
+	m := &Manager2{
+		mapper:       mapper,
+		templateRepo: templateRepo,
+		gp:           gp,
+	}
+	m.SetRuntimeState(rt)
+	return m
+}
+
+// Manager2 is the 0.2 adapter manager.
+type Manager2 struct {
+	// mapper is the selector and expression evaluator.
+	// It is passed through by Manager.
```

Comment:
Should we remove the second comment? At least to me it is unclear which Manager we are referring to.